src/lambdas/RedriveSNSDQLEvents/writeToFirehose.ts

```diff
@@ -0,0 +1,82 @@
+import { PutRecordBatchCommandOutput } from '@aws-sdk/client-firehose'
+import { firehosePutRecordBatch } from '../../sharedServices/firehose/firehosePutRecordBatch'
+import { logger } from '../../sharedServices/logger'
+import { AuditEvent } from '../../types/auditEvent'
+import { auditEventsToFirehoseRecords } from '../../utils/helpers/firehose/auditEventsToFirehoseRecords'
+import { getEnv } from '../../utils/helpers/getEnv'
+import { ProcessingResult } from './helper'
+
+export type FirehoseProcessingResult = {
+  successfullProcessingResults: ProcessingResult[]
+  failedProcessingResults: ProcessingResult[]
+}
+
+export const writeToFirehose = async (
+  successfullyParsedRecords: ProcessingResult[]
+): Promise<FirehoseProcessingResult> => {
+  const auditEventArray = successfullyParsedRecords.map((element) => {
+    return element.auditEvent as AuditEvent
+  })
+
+  const firehoseRecords = auditEventsToFirehoseRecords(auditEventArray)
+
+  try {
+    const result = await firehosePutRecordBatch(
+      getEnv('FIREHOSE_DELIVERY_STREAM_NAME'),
+      firehoseRecords
+    )
+    return parseFirehoseResponse(result, successfullyParsedRecords)
+  } catch (error) {
+    logger.error('failed to publish to firehose', error as Error)
+    const errorOutput: FirehoseProcessingResult = {
+      successfullProcessingResults: [],
+      failedProcessingResults: successfullyParsedRecords
+    }
+    return errorOutput
+  }
+}
+
+export const parseFirehoseResponse = (
```

Comment:
I think as a standard, we normally do one exported function per file, although not sure.

Reply:
sorted